### ISSUE TITLE
Invalid value for coindays destroyed in BCH block 498324

### ISSUE THREAD

**Issue by trippysalmon** [Blockchair/Blockchair.Support#41]
I noticed an invalid value for the coindays destroyed property.

The value seems to be `-2.4000300925926e-7` ([link](https://api.blockchair.com/bitcoin-cash/blocks?q=id(498324))) which of course is impossible.

Not that big of an issue but something you might want to look into.

Cheers

**Comment by Har01d** [comment]
Hello,

That's not a bug. We calculate the CDD values based on when blocks were mined. Sometimes (quite rarely) the next block comes "before" the previous, like in the case you provided:

Block height: 49832**4** time: 2017-10-27 17:28:_**23**_
Block height: 49832**5** time: 2017-10-27 17:28:_**09**_

That's a perfectly valid situation under the current consensus rules, because each miner has its own clock: https://bitcoin.org/en/developer-reference#block-headers

> The block time is a Unix epoch time when the miner started hashing the header (according to the miner). Must be strictly greater than the median time of the previous 11 blocks. Full nodes will not accept blocks with headers more than two hours in the future according to their clock.

Maybe we should add CDD values based on the median block time, this way there won't be negative CDD values.

Thank you!

**Comment by trippysalmon** [comment]
Yeah, I thought that might be the reason the CDD value was negative. I could have worded it differently than "invalid" because you are technically correct.

Using the median time would solve it indeed.

Anyway, good luck with your project!
